reazon.el

```diff
@@ -421,34 +421,47 @@ Also known as committed choice. This operator is impure."
        ((reazon--once ,head) ,@body)
        ,@rest))))
 
-(defmacro reazon-defrel (name varlist &rest goals)
-  "Define relation NAME with args VARLIST and body GOALS."
-  (declare (indent 2))
+(defmacro reazon-defrel (name varlist &optional docstring &rest goals)
+  "Define relation NAME with args VARLIST and body GOALS, and a DOCSTRING."
+  (declare (indent 2)
+           (doc-string 3))
   (let ((stream (gensym)))
-    `(defun ,name ,varlist
-       (lambda (,stream)
-         (lambda ()
-           (funcall (reazon-conj ,@goals) ,stream))))))
+    (if (stringp docstring)
```

Comment:
Is there a way to make it so that the macro logic isn't duplicated in the branches? What happens if an empty docstring is inserted?

Reply:
oops, that looks like a bug!